pkg/apply/taskrunner/task.go

```diff
@@ -232,6 +258,16 @@ func (w *WaitTask) StatusUpdate(taskContext *TaskContext, id object.ObjMetadata)
 	case w.skipped(taskContext, id):
 		// skipped - ignore
 		return
+	case w.stalled.Contains(id):
+		// If a stalled resource becomes current before other
+		// resources have completed/timed out, we consider it
+		// current.
+		if w.reconciledByID(taskContext, id) {
+			w.stalled = w.stalled.Remove(id)
+			w.sendEvent(taskContext, id, event.Reconciled)
+		}
+		// can't be all reconciled, so don't bother checking
```

Comment:
why can't they be all reconciled here?

Reply:
Updated the comment with an explanation.